storage/src/sign.rs

```diff
@@ -203,8 +206,13 @@ pub(crate) fn create_signed_buffer(
         header_names.join(";")
     };
 
-    let timestamp = now.to_rfc3339_opts(SecondsFormat::Secs, true).replace(['-', ':'], "");
-    let credential_scope = format!("{}/auto/storage/goog4_request", now.format("%Y%m%d"));
+    const CONFIG: EncodedConfig = well_known::iso8601::Config::DEFAULT.set_use_separators(false).encode();
+
+    let timestamp = now.format(&Iso8601::<CONFIG>).unwrap();
```

Comment:
For the `client::test::sign` and `sign::test::create_signed_buffer_test` errors, it seems that the `timestamp` format is the cause.
Try outputting seconds instead of nanoseconds.

The expected result of `create_signed_buffer`  is 
```
GOOG4-RSA-SHA256
20221212T135854Z     # <- second
20221212/auto/storage/goog4_request
5acaab81d2ecc3a1118e5e5af598b541c85e6e96a76544ca61bd4b6aa7351208
```

The actual result of `create_signed_buffer`  is 
```
GOOG4-RSA-SHA256
20221212T135723.395161000Z #  <- nanosecond
20221212/auto/storage/goog4_request
9d123e6c2c23ef7690e7346df14ae4c0e28839a14e446c870f8bddfbc428d8ef
```

Reply:
Thanks! That have been fixed now (I hope)